### ISSUE TITLE
Enabling computation of meds-evaluation metrics on the validation set each training epoch

### ISSUE THREAD

**Issue by Oufattole** [kamilest/meds-evaluation#9]
It seems the intended workflow is for meds-evaluation to be performed after a model is fully trained and has generated some labeled data. But I think we could wrap this in a pytorch lighting torchmetric and allow evaluation on the validation set every training epoch.

```python
import os
import tempfile
import subprocess
from typing import List, Dict
from datetime import datetime

import torch
from torchmetrics import Metric
import pyarrow as pa
import pyarrow.parquet as pq

class MEDSEvaluationMetric(Metric):
    def __init__(self):
        super().__init__()
        self.add_state("predictions", default=[], dist_reduce_fx="cat")
        self.temp_dir = tempfile.mkdtemp()
        self.output_file = os.path.join(self.temp_dir, "predictions.parquet")

    def update(self, preds: torch.Tensor, target: torch.Tensor, metadata: List[Dict]):
        batch_size = preds.shape[0]
        for i in range(batch_size):
            self.predictions.append({
                "subject_id": metadata[i]["subject_id"],
                "prediction_time": metadata[i]["prediction_time"],
                "boolean_value": bool(target[i].item()),
                "predicted_boolean_value": bool(preds[i].item() > 0.5),
                "predicted_boolean_probability": float(torch.sigmoid(preds[i]).item()),
            })

    def compute(self):
        # Create PyArrow table and write to Parquet file
        schema = pa.schema([
            ("subject_id", pa.int64()),
            ("prediction_time", pa.timestamp("us")),
            ("boolean_value", pa.bool_()),
            ("predicted_boolean_value", pa.bool_()),
            ("predicted_boolean_probability", pa.float64()),
        ])
        
        table = pa.Table.from_pylist(self.predictions, schema=schema)
        pq.write_table(table, self.output_file)

        # Run meds-evaluation subprocess
        cmd = f"meds-evaluation input_dir={self.temp_dir}"
        try:
            result = subprocess.run(cmd, shell=True, check=True, capture_output=True, text=True)
            print("meds-evaluation output:", result.stdout)
        except subprocess.CalledProcessError as e:
            print("Error running meds-evaluation:", e)
            print("Error output:", e.output)

        # Clean up temporary directory
        for file in os.listdir(self.temp_dir):
            os.remove(os.path.join(self.temp_dir, file))
        os.rmdir(self.temp_dir)

        # Return None as the actual metrics are computed by the meds-evaluation script
        return None

    def reset(self):
        super().reset()
        self.predictions = []
        self.temp_dir = tempfile.mkdtemp()
        self.output_file = os.path.join(self.temp_dir, "predictions.parquet")
```

**Comment by mmcdermott** [comment]
I think this issue is a viable extension of this package (e.g., for tracking an entire MEDS-evaluation report during training), but I think that (a) is a low priority change that would add unnecessary complexity to the tool as during training often you want a different set of metrics than what MEDS-evaluation reports and (b) I think the manner in which you indicated it could be solved is likely not the right approach here @Oufattole. Instead, we'd need to use dedicated `torchmetrics` metrics for each of the tracked components. A different approach that could be viable (but would be handled outside of this package) would be to set up a Lightning callback to collect predictions and run this package every N iterations on the validation set, though that would be expensive. But for now, I'm going to mark this as low priority as I don't think it will be something we'll work on in the foreseeable future.
